src/MoBi.Presentation/Presenter/EditFavoritesPresenter.cs

```diff
@@ -1,102 +1,73 @@
 using OSPSuite.Core.Events;
 using OSPSuite.Core.Services;
-using OSPSuite.Presentation.Core;
-using OSPSuite.Presentation.DTO;
 using OSPSuite.Presentation.Presenters;
-using OSPSuite.Presentation.Presenters.ContextMenus;
-using OSPSuite.Utility;
 using OSPSuite.Utility.Collections;
 using OSPSuite.Utility.Events;
 using OSPSuite.Utility.Extensions;
 
 namespace MoBi.Presentation.Presenter
 {
-   public interface IEditFavoritesPresenter : IEditParameterListPresenter,
+   public interface IEditFavoritesPresenter : IPresenter,
       IListener<AddParameterToFavoritesEvent>,
       IListener<RemoveParameterFromFavoritesEvent>,
       IListener<FavoritesLoadedEvent>,
+      IListener<FavoritesOrderChangedEvent>,
       IListener<AddedEvent>,
       IListener<RemovedEvent>
    {
-      void Select(ParameterDTO parameterDTO);
       Func<IObjectBase, bool> ShouldHandleRemovedEvent { set; }
+      void MoveUp();
+      void MoveDown();
+
    }
 
    public interface IEditFavoritesPresenter<T> : IEditFavoritesPresenter, IEditPresenter<T> where T : IObjectBase
    {
    }
 
-   public abstract class EditFavoritesPresenter<T> : AbstractParameterBasePresenter<IEditParameterListView, IEditParameterListPresenter>,
+   public abstract class EditFavoritesPresenter<T> : AbstractCommandCollectorPresenter<IEditFavoritesView, IEditFavoritesPresenter>,
       IEditFavoritesPresenter<T> where T : IObjectBase
    {
-      private readonly IParameterToParameterDTOMapper _parameterDTOMapper;
-      protected readonly List<ParameterDTO> _favorites;
       private readonly IFavoriteRepository _favoriteRepository;
       protected Cache<string, IParameter> _parameterCache;
       protected readonly IEntityPathResolver _entityPathResolver;
-      private readonly IMoBiContext _context;
-      private readonly IViewItemContextMenuFactory _viewItemContextMenuFactory;
+      protected readonly IEditParameterListPresenter _editParameterListPresenter;
+      private readonly IFavoriteTask _favoriteTask;
       protected T _projectItem;
+
       public Func<IObjectBase, bool> ShouldHandleRemovedEvent { protected get; set; }
 
-      protected EditFavoritesPresenter(IEditParameterListView view, IQuantityTask quantityTask,
-         IInteractionTaskContext interactionTaskContext, IFormulaToFormulaBuilderDTOMapper formulaMapper,
-         IParameterToParameterDTOMapper parameterDTOMapper, IFavoriteRepository favoriteRepository,
-         IInteractionTasksForParameter parameterTask, IFavoriteTask favoriteTask, IEntityPathResolver entityPathResolver, IViewItemContextMenuFactory viewItemContextMenuFactory)
-         : base(
-            view, quantityTask, interactionTaskContext,
-            formulaMapper, parameterTask, favoriteTask)
+      protected EditFavoritesPresenter(
+         IEditFavoritesView view,
+         IFavoriteRepository favoriteRepository,
+         IEntityPathResolver entityPathResolver,
+         IEditParameterListPresenter editParameterListPresenter,
```

Comment:
We did not have any stand alone implementation of the EditParameterListPresenter... now we do :)